src/metagen/src/client_rs/static/client.rs

```diff
@@ -1250,175 +1439,423 @@ pub mod graphql {
     // PlaceholderValue, fieldName -> gql_var_name
     type FoundPlaceholders = Vec<(PlaceholderValue, HashMap<CowStr, CowStr>)>;
 
-    fn select_node_to_gql(
-        ty_to_gql_ty_map: &TyToGqlTyMap,
-        dest: &mut impl std::fmt::Write,
-        node: SelectNodeErased,
-        variable_types: &mut HashMap<CowStr, CowStr>,
-        variables_object: &mut JsonObject,
-        placeholders: &mut FoundPlaceholders,
-    ) -> std::fmt::Result {
-        if node.instance_name != node.node_name {
-            write!(dest, "{}: {}", node.instance_name, node.node_name)?;
-        } else {
-            write!(dest, "{}", node.node_name)?;
-        }
-        if let Some(args) = node.args {
-            match args {
-                NodeArgsMerged::Inline(args) => {
-                    if !args.is_empty() {
-                        write!(dest, "(")?;
-                        for (key, val) in args {
-                            let name = format!("in{}", variable_types.len());
-                            write!(dest, "{key}: ${name}, ")?;
-                            variables_object.insert(name.clone(), val.value);
-                            variable_types.insert(name.into(), val.type_name);
-                        }
-                        write!(dest, ")")?;
+    #[derive(Debug)]
+    struct FileExtractor {
```

Comment:
nitpick: Maybe move this construct out of the `graphql` mod? It doesn't seem transport specific.